### ISSUE TITLE
make directory order wrong??

### ISSUE THREAD

**Issue by yaobinshi** [fossology/fossology#333]
---

Author Name: **Bob Gobeille** 
Original Redmine Issue: 7479, http://www.fossology.org/issues/7479
Original Date: 2014/08/21
Original Assignee: Dong Ma

---

This is a snippet from doing a make in a source tree.  Notice how there are repeated makes in src/lib/c and libfossology.a.  This looks like a bug.   

```
make -s -C copyright
make[2]: Entering directory `/home/bobg/fossology/src/copyright'
make[3]: Entering directory `/home/bobg/fossology/src/copyright/agent'
make[4]: Entering directory `/home/bobg/fossology/src/lib/c'
make libfossology.a
make[4]: Leaving directory `/home/bobg/fossology/src/lib/c'
make copyright agent
make[3]: Leaving directory `/home/bobg/fossology/src/copyright/agent'
make[3]: Entering directory `/home/bobg/fossology/src/copyright/ui'
make copyright ui
make[3]: Leaving directory `/home/bobg/fossology/src/copyright/ui'
make[2]: Leaving directory `/home/bobg/fossology/src/copyright'
make -s -C lib
make[2]: Entering directory `/home/bobg/fossology/src/lib'
make[3]: Entering directory `/home/bobg/fossology/src/lib/c'
make libfossology.a
make[3]: Leaving directory `/home/bobg/fossology/src/lib/c'

```

**Comment by yaobinshi** [comment]
---

Original Redmine Comment
Author Name: **Dong Ma** 
Original Date: 2014/08/21 15:43:21 +0000

---

The first time of make src/lib/c is because when make copyright agent it need src/lib/c to be built first.
The second time of make src/lib/c is make the lib directory.

If we need to change this behavior?

**Comment by yaobinshi** [comment]
---

Original Redmine Comment
Author Name: **Bob Gobeille** 
Original Date: 2014/08/21 16:26:57 +0000

---

I don't understand your explanation.  When you do a make in lib/c it builds the library.  I only included a snippet from the make output.  I'm attaching the whole output (make.out) so you can see what I mean.  There are four "make libfossology.a".

moving to 2.7

**Comment by pguttmann** [comment]
Attachment - http://www.fossology.org/attachments/4365/make.out

**Comment by mcjaeger** [comment]
moving it to 3.0.1 because I understand this is about redundancy and reomving it...?
